tools/audit_policy/main.go

```diff
@@ -71,104 +141,106 @@ func cwd() string {
+				return s.Name, "", nil
 			}
+			return "", "", err
+		}
+		s.Namespace = strings.Trim(string(ns), "\r\n")
+	}
 
-			if name, ok := chart["name"]; ok {
-				chartNames[modulePath] = name.(string)
-			}
+	return s.Name, s.Namespace, nil
+}
 
+func processServiceAccounts(templatesDir string) ([]string, error) {
+	serviceAccounts := make([]string, 0)
+	err := filepath.Walk(templatesDir, func(path string, f os.FileInfo, err error) error {
+		if f != nil && f.IsDir() {
 			return nil
 		}
 
-		if filepath.Base(path) == ".namespace" {
-			ns, err := os.ReadFile(path)
-			if err != nil {
-				return err
-			}
-			namespace = strings.Trim(string(ns), "\r\n")
-
-			if !strings.HasPrefix(namespace, "d8-") && namespace != "kube-system" {
-				return nil
-			}
+		if filepath.Base(path) != "rbac-for-us.yaml" {
+			return nil
+		}
 
-			*namespaces = append(*namespaces, namespace)
+		rbac, err := os.ReadFile(path)
+		if err != nil {
+			return err
 		}
 
-		if filepath.Base(path) == "rbac-for-us.yaml" {
-			rbac, err := os.ReadFile(path)
-			if err != nil {
-				return err
-			}
-			r := regexp.MustCompile(`apiVersion: v1
+		r := regexp.MustCompile(`apiVersion: v1
 kind: ServiceAccount
 metadata:
 \s*name:\s*(.*)
 `)
-			match := r.FindAllStringSubmatch(string(rbac), -1)
-			for _, m := range match {
-				// ServiceAccount could be in double or single quotes
-				// Unquote it for versatility
-				saName, err := unquote(m[1])
+		match := r.FindAllStringSubmatch(string(rbac), -1)
+		for _, m := range match {
+			saName := strings.TrimSpace(m[1])
+			if strings.HasPrefix(saName, "\"") {
+				saName, err = unquote(saName)
 				if err != nil {
 					return err
 				}
-
-				// ServiceAccounts for istiod will be added to audit rules in
-				// modules/040-control-plane-manager/hooks/audit_policy.go
-				if strings.HasPrefix(saName, "istiod-") {
-					continue
-				}
-
-				saNames[modulePath] = append(
-					saNames[modulePath],
-					fmt.Sprintf("system:serviceaccount:%s:%s", namespace, saName),
-				)
 			}
+
+			serviceAccounts = append(serviceAccounts, saName)
 		}
 
 		return nil
+
 	})
 
-	for modulePath, saNames := range saNames {
-		for _, sa := range saNames {
-			chartName := ""
-			if name, ok := chartNames[modulePath]; ok {
-				chartName = name
-			}
-			sa = strings.Replace(sa, "{{ .Chart.Name }}", chartName, 1)
-			sa = strings.Replace(sa, "{{ $.Chart.Name }}", chartName, 1)
-			if len(sa) == 0 && len(chartName) == 0 {
-				return fmt.Errorf("empty final SA name, seems chartName didnt resolve for module: %s", modulePath)
-			}
-			*sas = append(*sas, sa)
+	return serviceAccounts, err
+}
+
+func handleServiceAccounts(templatesDir string, modulePath string, name string, ns string, sas map[string]struct{}) error {
+	serviceAccounts, err := processServiceAccounts(templatesDir)
+	if err != nil {
+		return err
+	}
+
+	for _, saName := range serviceAccounts {
+		// ServiceAccounts for istiod will be added to audit rules in
```

Comment:
will NOT?

because in the next line we continue and skip processing

Reply:
Have no idea about this logic, honestly. It's just a paste from the [current file](https://github.com/deckhouse/deckhouse/blob/main/tools/audit_policy/main.go#L140-L144)